File: java/javadpbo/src/com/sem4dpbo/week3/praktikum3/latihanpraktikum3_java/Processor.java
package com.sem4dpbo.week3.praktikum3.latihanpraktikum3_java;

public class Processor {
    private String name;
    private Long procPrice;

    //constructor
    Processor() {}
    Processor(String name, long price) {
        this.name = name;
        this.procPrice = price;
    }

    //set and get name
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    //set and get price
    public Long getProcPrice() {
        return procPrice;
    }

    public void setProcPrice(Long procPrice) {
        this.procPrice = procPrice;
    }

    public void printProcessor(){
        System.out.println("-Data Processor");
        System.out.println(" Name            : "+this.name);
        System.out.println(" Processor Price : "+this.procPrice);
        System.out.println("------------------------------------");
    }
}

File: cpp/3rd_Week/Praktikum/latihanpraktikum3_cpp/Processor.cpp
#include <iostream>
#include <string>

using namespace std;

class Processor
{
private:
    /* data */
    string name;
    long procPrice;
public:
    //contrutor
    Processor() {}
    Processor(string name, long price) {
        this->name = name;
        this->procPrice = price;
    }

    // set and get atribut
    void setName(string n){ this->name = n; }
    void setProcPrice(long p){ this->procPrice = p; }
    
    string getName(){ return this->name; }
    long getProcPrice(){ return this->procPrice; }

    //method print
    void printProcessor(){
        cout << "-Data Proccesor" << endl;
        cout << " Name            : " << this->name << endl;
        cout << " Proccesor Price : " << this->procPrice << endl;
        cout << "-----------------------------------" << endl;
    }
    ~Processor() {}
};